### ISSUE TITLE
Transparency regressions for hard objects

### ISSUE THREAD

**Issue by andrewkchan** [NJU-3DV/Relightable3DGaussian#17]
Hi, thanks for sharing your technique, it looks very promising!

I am seeing some issues with hard objects noticeably being transparent when they are not in models trained by other approaches*. For example the table in the MipNeRF360 "garden" scene appears transparent in a model trained by this approach but not in the original.

Viewed from the side (notice the wooden legs and the bottom of the table) - original first, R3DG second:
<img width="660" alt="vanilla_garden_table_opacityfromside" src="https://github.com/NJU-3DV/Relightable3DGaussian/assets/8591901/81c1ad56-b245-4ade-b5f2-68f0876eefd7">
<img width="660" alt="r3dg_garden_table_opacityfromside" src="https://github.com/NJU-3DV/Relightable3DGaussian/assets/8591901/959051d2-f1dc-457c-90a8-9d89d851922f">

Viewed from the top (notice you can see-through to the floor as the camera pans):

https://github.com/NJU-3DV/Relightable3DGaussian/assets/8591901/c1661095-1e12-4694-9a91-648b9054dcae

https://github.com/NJU-3DV/Relightable3DGaussian/assets/8591901/e613528f-b737-448a-8088-4ada78a80d47

*Actually I'm not sure if the original model ([source](https://huggingface.co/cakewalk/splat-data/tree/main)) was trained with the original 3DGS code or with an approach like SparseGS, or what the hyper-parameters were. But it would still be helpful to understand if this is a failure mode with known remedies. For example, is it possible that I created the wrong depth/mask images or that I need to tweak a hyperparameter?

**Comment by JiatengLiu** [comment]
same issue
